### ISSUE TITLE
Force BERTopic to create N topics 

### ISSUE THREAD

**Issue by TAsUjxnMIL** [MaartenGr/BERTopic#443]
Hello Maarten,
is it possible to force BERTopic to create a certain number of topics? 
In a specific corpus only 2 topics are generated but the corpus has 3 main topics. So I would like to force BERTopic to create 3 topics.
Is there any possibility to do that? 

Thank you very much!

**Comment by MaartenGr** [comment]
Typically, if you have less than 10 topics generated, there is typically too little data available to create good topic representations. I would advise lowering `min_topic_size` to generate more topics or, if at all possible, add more data.

**Comment by TAsUjxnMIL** [comment]
Okay that means there is no possibility to force BERTopic to create a certain number of topics?

**Comment by MaartenGr** [comment]
Closing this for now. However, if you want to discuss this further, let me know and I'll make sure to reopen the issue!

**Comment by hari-chalise** [comment]
![image](https://github.com/user-attachments/assets/fbac7fd6-9add-43c1-bc4e-d4ea85b83d28)
 I have the same Problem Only 2 topic are generated  these are also not appropriate topic. What can I do? Thanks In Advance.
This is code Line
`from umap import UMAP
umap_model = UMAP(n_neighbors=15, n_components= 5, min_dist = 0.1, metric = 'cosine', random_state = 42)
umap_model

from hdbscan import HDBSCAN
hdbscan_model = HDBSCAN(min_cluster_size=20, metric='euclidean', cluster_selection_method='eom', prediction_data= True)
hdbscan_model

from sklearn.feature_extraction.text import CountVectorizer
nepali_stop_words =  [
    "एउटा", "अर्को", "आइपुग्छ", "आफु", "आदि", "उन", "उनको", "उनिहरु", "उनै", "उक्त", 
    "एक", "केही", "को", "किन", "कै", "कसै", "कतै", "गर्दा", "गरेर", "गर्यो", "गर्छ", "गर्नु", 
    "गरिन्छ", "गरे", "भए", "भएन", "भएपछी", "भएको", "हुनु", "हुन्छ", "हो", "हुन", "छन्", "छ",
    "ठिक", "त्यस", "त्यसमा", "त्यहाँ", "त्यही", "ति", "तिनी", "तिनको", "तिनलाई", "तपाई", "तपाईको",
    "त्यो", "देखि", "दिनुहोस्", "दिन्छ", "दिनु", "पनि", "पुन", "पछि", "पट्टि", "पर", "पर्दछ", "पुग्न", "पुग्यो",
    "पुर्ण", "पुरै", "प्रतिकुल", "प्राय", "प्रायः", "पुर्णतः", "र", "रह्यो", "रहन", "रहनु", "रहेका", "रहन्छ", "रहे",
    "रहेछन्", "रहने", "लगायत", "ले", "लिएर", "लिन", "लिनु", "लिए", "लगाएत", "सहित", "सँग", "सँगै", "संग",
    "संगसंगै", "संग", "साथै", "सब", "सबै", "सकिन्छ", "सकेसम्म", "सकेको", "सकेपछि", "सके", "सहित", "समेत", 
    "मा", "मात्र", "माथि", "मुनि", "मेरो", "यदी", "यदि", "यही" "गर" , "भन", "पन", "सम" "गर", "रक", "हर", "छन", "सक",

    # Add more Nepali stop words as needed
]
vectorizer_model= CountVectorizer(stop_words = nepali_stop_words, min_df=1, max_df=3, ngram_range =(1,8))
vectorizer_model

from bertopic import BERTopic
topic_model = BERTopic(
    embedding_model = model,
    umap_model= umap_model,
    hdbscan_model= hdbscan_model,
    vectorizer_model= vectorizer_model,
    # nr_topics = "auto",
    min_topic_size=20,
    # stop_words = nepali_stop_words,
    # diversity=0.7,
    #hyperparameter
    top_n_words= 5,
    verbose=True
)
topics, probs=topic_model.fit_transform(documents, corpus_embedding) `

**Comment by MaartenGr** [comment]
@hari-chalise That is likely a result of setting the `min_cluster_size` to high. If you lower it quite a bit (e.g., 5), then you are likely to have more topics.

**Comment by hari-chalise** [comment]
Thanks @MaartenGr its really helpful your response.

![Image](https://github.com/user-attachments/assets/bff37e36-7b2f-4567-8b5b-63f754af2023)

But i have also struggling to find unique words in topic. There is same word repeated in same i.e आयोग, आयोगको, आयोगलाई. Is any suggestion for this?

**Comment by MaartenGr** [comment]
@hari-chalise Use KeyBERTInspired with an embedding model that is appropriate for your language. That will likely remove words with similar meanings as long as the embedding model is trained in the same language as your data.
